edifice/engine.py

```diff
@@ -438,20 +442,29 @@ def traverse(comp, parent):
         # 2) For all such old components, construct a new component and merge in old component props
         for parts in components_to_replace:
             old_comp, new_comp_class, _, _ = parts
-            parameters = list(inspect.signature(new_comp_class.__init__).parameters.items())
-
-            try:
-                kwargs = {k: old_comp.props[k] for k, v in parameters[1:]
-                          if v.default is inspect.Parameter.empty and k[0] != "_"}
-            except KeyError:
-                k = None
-                for k, _ in parameters[1:]:
-                    if k not in old_comp.props:
-                        break
-                raise ValueError(
-                    f"Error while reloading {old_comp}: "
-                    f"New class expects prop ({k}) not present in old class"
-                )
+            if 'kwargs' in inspect.signature(new_comp_class.__init__).parameters:
+                # Then new_comp_class is a new-style @component Element.
+                kwargs = old_comp._props.get('kwargs', {})
```

Comment:
This doesn't seem right to me. I think the old component should get the props directly passed to it, not wrapped in a `kwargs` prop. If this is the case I think there's a bug elsewhere.

Reply:
https://github.com/pyedifice/pyedifice/blob/a8eec55c2c06f7df10f5fc6b2862b1c7e215e55a/edifice/_component.py#L660

Couple of notes:
- The above is missing all `*args`.
- We should probably use `func_tools.wrap` to have the `__init__` method get the correct parameters? (Will this survive to inspection, though?)

Reply:
See the comment below

https://github.com/pyedifice/pyedifice/blob/a1b4ddfd418a87467d9bdd421ee761e2520fa90a/edifice/engine.py#L464-L468